yoshi/pkg/controllers/vm_controller.go

```diff
@@ -0,0 +1,302 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	plumberv1 "github.com/platform9/luigi/yoshi/api/v1"
+	"github.com/platform9/luigi/yoshi/pkg/utils/constants"
+	"github.com/platform9/luigi/yoshi/pkg/utils/iputils"
+	"github.com/platform9/luigi/yoshi/pkg/utils/vmutils"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	kubevirtv1 "kubevirt.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// NetworkWizardReconciler reconciles a NetworkWizard object
+type VMReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Log    logr.Logger
+}
+
+type VMReqWrapper struct {
+	Log         logr.Logger
+	Client      client.Client
+	needsUpdate bool
+	vm          *kubevirtv1.VirtualMachine
+	networks    []*plumberv1.NetworkWizard
+}
+
+func NewVMReqWrapper(log logr.Logger, client client.Client) *VMReqWrapper {
+	reqInfo := new(VMReqWrapper)
+	reqInfo.Log = log
+	reqInfo.Client = client
+	reqInfo.needsUpdate = false
+	return reqInfo
+}
+
+func (req *VMReqWrapper) WithVM(vm *kubevirtv1.VirtualMachine) *VMReqWrapper {
+	req.vm = vm
+	return req
+}
+
+func (req *VMReqWrapper) WithNetworks(networks ...*plumberv1.NetworkWizard) *VMReqWrapper {
+	if req.networks == nil {
+		req.networks = []*plumberv1.NetworkWizard{}
+	}
+	req.networks = append(req.networks, networks...)
+	return req
+}
```

Comment:
Nit: maybe move this down, since it is now in between the VMReconciler (method) definition